src/consensus/pbft_manager.cpp

```diff
@@ -1059,131 +1050,19 @@ void PbftManager::secondFinish_() {
-    } else if (v.getRound() != vote_round) {
-      LOG(log_er_) << "Vote has a different round with " << vote_round << ". VOTE: " << v;
-      assert(false);
-    } else if (v.getStep() != vote_step) {
-      LOG(log_er_) << "Next phase vote has a different step with " << vote_step << ". VOTE: " << v;
-      assert(false);
-    }
-
-    auto voted_block_hash = v.getBlockHash();
-    if (blockhash_votes_map.count(voted_block_hash)) {
-      blockhash_votes_map[voted_block_hash].emplace_back(v);
-    } else {
-      std::vector<Vote> voted_block_hash_votes{v};
-      blockhash_votes_map[voted_block_hash] = voted_block_hash_votes;
-    }
-
-    for (auto const &blockhash_votes : blockhash_votes_map) {
-      if (blockhash_votes.second.size() == TWO_T_PLUS_ONE) {
-        LOG(log_dg_) << "Find voted block hash " << blockhash_votes.first << " vote type " << unsigned(vote_type)
-                     << " in round " << vote_round << " step " << vote_step << " has " << blockhash_votes.second.size()
-                     << " votes";
-        return votesBundle(true, blockhash_votes.first, blockhash_votes.second);
-      } else {
-        LOG(log_tr_) << "Don't have enough votes. voted block hash " << blockhash_votes.first << " vote type "
-                     << unsigned(vote_type) << " for round " << vote_round << " step " << vote_step << " has "
-                     << blockhash_votes.second.size() << " votes (2TP1 = " << TWO_T_PLUS_ONE << ")";
-      }
-    }
-  }
-
-  return votesBundle();
-}
-
-std::vector<Vote> PbftManager::getVotesOfTypeFromVotesForRoundAndStep_(PbftVoteTypes vote_type,
-                                                                       std::vector<Vote> &votes, uint64_t round,
-                                                                       size_t step,
-                                                                       std::pair<blk_hash_t, bool> blockhash) {
-  std::vector<Vote> votes_of_requested_type;
-  std::copy_if(votes.begin(), votes.end(), std::back_inserter(votes_of_requested_type),
-               [vote_type, round, step, blockhash](Vote const &v) {
-                 return (v.getType() == vote_type && v.getRound() == round && v.getStep() == step &&
-                         (blockhash.second == false || blockhash.first == v.getBlockHash()));
-               });
-
-  return votes_of_requested_type;
-}
-
-Vote PbftManager::generateVote(blk_hash_t const &blockhash, PbftVoteTypes type, uint64_t round, size_t step,
-                               size_t weighted_index) {
+std::shared_ptr<Vote> PbftManager::generateVote(blk_hash_t const &blockhash, PbftVoteTypes type, uint64_t round,
+                                                size_t step, size_t weighted_index) {
   // sortition proof
   VrfPbftMsg msg(type, round, step, weighted_index);
   VrfPbftSortition vrf_sortition(vrf_sk_, msg);
   Vote vote(node_sk_, vrf_sortition, blockhash);
 
-  return vote;
+  return std::make_shared<Vote>(vote);
```

Comment:
retturn std::make_shared<Vote>(node_sk_, vrf_sortition, blockhash) and delete line 1058

Reply:
Fixed, thanks